jujugui/static/gui/src/app/components/basic-table/test-basic-table.js

```diff
@@ -0,0 +1,83 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+var juju = {components: {}}; // eslint-disable-line no-unused-vars
+
+chai.config.includeStack = true;
+chai.config.truncateThreshold = 0;
+
+describe('BasicTable', function() {
+  beforeAll(function(done) {
+    // By loading this file it adds the component to the juju components.
+    YUI().use('basic-table', function() { done(); });
+  });
+
+  it('can render the table', function() {
+    const columns = [{
+      title: 'Column 1',
+      size: 3,
+      classes: ['class1', 'class2']
+    }, {
+      title: 'Column 2',
+      size: '4'
+    }];
+    const rows = [
+      ['row 1 column 1', 'row 1 column 2'],
+      ['row 2 column 1'],
+      ['row 3 column 1', 'row 3 column 2', 'row 3 column 3']
+    ];
+    const renderer = jsTestUtils.shallowRender(
+      <juju.components.BasicTable
+        columns={columns}
+        rows={rows} />, true);
+    const output = renderer.getRenderOutput();
+    const expected = (
+      <ul className="basic-table twelve-col">
```

Comment:
We really don't use tables for anything, it is rarely used at all in HTML these days (see my explanation above).